src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.CacheEntry.cs

```diff
@@ -0,0 +1,150 @@
+                    foreach (var (targetType, symbolInfoIndices) in syntaxIndex.SimpleExtensionMethodInfo)
+                    {
+                        foreach (var index in symbolInfoIndices)
+                        {
+                            _simpleItemBuilder.Add(targetType, syntaxIndex.DeclaredSymbolInfos[index]);
+                        }
+                    }
+
+                    foreach (var index in syntaxIndex.ComplexExtensionMethodInfo)
+                    {
+                        _complexItemBuilder.Add(syntaxIndex.DeclaredSymbolInfos[index]);
+                    }
+                }
+
+                public void Dispose()
+                    => _complexItemBuilder.Free();
+            }
+        }
+
+        /// <summary>
+        /// We don't use PE cache from the service, so just pass in type `object` for PE entries.
+        /// </summary>
+        [ExportWorkspaceServiceFactory(typeof(IImportCompletionCacheService<CacheEntry, object>), ServiceLayer.Editor), Shared]
+        private sealed class CacheServiceFactory : AbstractImportCompletionCacheServiceFactory<CacheEntry, object>
+        {
+        }
+
+        private static IImportCompletionCacheService<CacheEntry, object> GetCacheService(Workspace workspace)
+            => workspace.Services.GetRequiredService<IImportCompletionCacheService<CacheEntry, object>>();
+
+        private static async Task<CacheEntry?> GetCacheEntryAsync(
+            Project project,
+            bool loadOnly,
+            IImportCompletionCacheService<CacheEntry, object> cacheService,
+            CancellationToken cancellationToken)
+        {
+            // While we are caching data from SyntaxTreeInfo, all the things we cared about here are actually based on sources symbols.
+            // So using source symbol checksum would suffice.
+            var checksum = await SymbolTreeInfo.GetSourceSymbolsChecksumAsync(project, cancellationToken).ConfigureAwait(false);
+
+            // Cache miss, create all requested items.
+            if (!cacheService.ProjectItemsCache.TryGetValue(project.Id, out var cacheEntry) ||
+                cacheEntry.Checksum != checksum ||
+                cacheEntry.Language != project.Language)
+            {
+                var syntaxFacts = project.LanguageServices.GetRequiredService<ISyntaxFactsService>();
+                using var builder = new CacheEntry.Builder(checksum, project.Language, syntaxFacts.StringComparer);
+
+                foreach (var document in project.Documents)
+                {
+                    // Don't look for extension methods in generated code.
+                    if (document.State.Attributes.IsGenerated)
+                    {
+                        continue;
+                    }
+
+                    var info = await document.GetSyntaxTreeIndexAsync(loadOnly, cancellationToken).ConfigureAwait(false);
+                    if (info == null)
+                    {
+                        return null;
```

Comment:
please doc that this is an intentional decision, and you didn't accidentally forget to 'continue' here.